### PR TITLE
Adds `variant` and `interactive` props to `gr.UploadButton`

### PR DESCRIPTION
**Pull request by abidlabs**
Closes: #4364 

Test code:

```py
import gradio as gr

with gr.Blocks() as demo:
    gr.Button()
    gr.Button(interactive=False)
    gr.Button(variant="primary")
    gr.Button(variant="stop", interactive=False)
    gr.UploadButton()
    gr.UploadButton(interactive=False)
    gr.UploadButton(variant="primary")
    gr.UploadButton(variant="stop", interactive=False)
```

image

### CODE REVIEW

File: js/app/src/components/UploadButton/UploadButton.svelte [hunk 1]
```diff
@@ -16,6 +16,9 @@
 	export let file_count: string;
 	export let file_types: Array<string> = ["file"];
 	export let root: string;
+	export let mode: "static" | "dynamic" = "dynamic";
+	export let variant: "primary" | "secondary" | "stop" = "secondary";
+	console.log("mode", mode);
```
Comment (dawoodkhan82) [inline js/app/src/components/UploadButton/UploadButton.svelte:21]: remove console.log

**Review by dawoodkhan82 — CHANGES_REQUESTED** [review]

**Comment by dawoodkhan82** [comment]
Seems Like the UploadButton is still interactive even when set to false.

**Comment by dawoodkhan82** [comment]
Also what is the purpose of adding variants?

File: js/app/src/components/UploadButton/UploadButton.svelte (hunk 1 above)
Comment (abidlabs) [reply js/app/src/components/UploadButton/UploadButton.svelte]: Whoops fixed

**Comment by abidlabs** [comment]
> Also what is the purpose of adding variants?

Just for further customization. I imagine someone might want primary/secondary at least. 

> Seems Like the UploadButton is still interactive even when set to false.

Really? Is this after the frontend has been rebuilt? Here's what I get testing the code above:

image

**Comment by dawoodkhan82** [comment]
works for me, didn't have the latest changes pulled

**Comment by dawoodkhan82** [comment]
> Just for further customization. I imagine someone might want primary/secondary at least.
Should we remove `stop` as a variant? Seems confusing for this component

Otherwise lgtm!

**Comment by abidlabs** [comment]
> Should we remove stop as a variant? Seems confusing for this component

I think it's better to make the API between `Button` and `UploadButton` as consistent as possible. It's fine if the "stop" variant is not really used, but it might turn out to be useful in styling. 

Thanks for the review @dawoodkhan82!

**Pull request merged**